src/lib/core/WeaveMessageLayer.h

```diff
@@ -827,14 +857,14 @@ inline uint64_t WeaveFabricIdToIPv6GlobalId(uint64_t fabricId) { return (fabricI
  */
 inline uint64_t IPv6InterfaceIdToWeaveNodeId(uint64_t interfaceId)
 {
-    return (interfaceId & ~(0x0200000000000000ULL));
+    return (interfaceId <= kMaxAlwaysLocalWeaveNodeId) ? interfaceId : (interfaceId ^ kEUI64_UL_Mask);
 }
 
 /**
  *  Convert a Weave node identifier to an IPv6 address interface identifier.
  *
- *  Weave node identifiers are global EUI-64s, which per RFC-3513 are converted to interface
- *  identifiers by setting the universal/local bit to 1 (bit 57 counting the LSB as 0).
+ *  Weave node identifiers are Universal/Local EUI-64s, which per RFC-3513 are converted to
+ *  interface identifiers by inverting the universal/local bit (bit 57 counting the LSB as 0).
  *
  *  As a convenience to testing, node identifiers less than 65536 are considered 'local',
```

Comment:
It might be helpful to the reader of comments-only to include the mnemonic `#kMaxAlwaysLocalWeaveNodeId` alongside the "less than 65536".

Reply:
done